SQL/Release_patches/16.1_To_17.0_upgrade.sql

```diff
@@ -0,0 +1,225 @@
+  CONSTRAINT `fk_issues_5` FOREIGN KEY (`CenterID`) REFERENCES `psc` (`CenterID`),
+  CONSTRAINT `fk_issues_6` FOREIGN KEY (`lastUpdatedBy`) REFERENCES `users` (`UserID`)
+) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8;
+
+-- issues_history table
+CREATE TABLE `issues_history` (
+  `issueHistoryID` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `newValue` longtext NOT NULL,
+  `dateAdded` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `fieldChanged` enum('assignee','status','comment','sessionID','centerID','title','category','module','lastUpdatedBy','priority') NOT NULL DEFAULT 'comment',
+  `issueID` int(11) unsigned NOT NULL,
+  `addedBy` varchar(255) NOT NULL DEFAULT '',
+  PRIMARY KEY (`issueHistoryID`),
+  KEY `fk_issues_comments_1` (`issueID`),
+  CONSTRAINT `fk_issues_comments_1` FOREIGN KEY (`issueID`) REFERENCES `issues` (`issueID`)
+) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8;
+
+-- issues_comments table
+CREATE TABLE `issues_comments` (
+  `issueCommentID` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `issueID` int(11) unsigned NOT NULL,
+  `dateAdded` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `addedBy` varchar(255) NOT NULL DEFAULT '',
+  `issueComment` text NOT NULL,
+  PRIMARY KEY (`issueCommentID`),
+  KEY `fk_issue_comments_1` (`issueID`),
+  CONSTRAINT `fk_issue_comments_1` FOREIGN KEY (`issueID`) REFERENCES `issues` (`issueID`)
+) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8;
+
+-- issues_comments_history
+CREATE TABLE `issues_comments_history` (
+  `issueCommentHistoryID` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `issueCommentID` int(11) unsigned NOT NULL,
+  `dateEdited` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `newValue` longtext NOT NULL,
+  `editedBy` varchar(255) NOT NULL DEFAULT '',
+  PRIMARY KEY (`issueCommentHistoryID`),
+  KEY `fk_issues_comments_history` (`issueCommentID`),
+  CONSTRAINT `fk_issues_comments_history` FOREIGN KEY (`issueCommentID`) REFERENCES `issues_comments` (`issueCommentID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- issues_watching
+CREATE TABLE `issues_watching` (
+  `userID` varchar(255) NOT NULL DEFAULT '',
+  `issueID` int(11) unsigned NOT NULL,
+  PRIMARY KEY (`userID`,`issueID`),
+  KEY `fk_issues_watching_2` (`issueID`),
+  CONSTRAINT `fk_issues_watching_1` FOREIGN KEY (`userID`) REFERENCES `users` (`UserID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+UPDATE files_qcstatus as fqc SET fqc.Selected='true' WHERE fqc.Selected <> '';
+ALTER TABLE files_qcstatus CHANGE `Selected` `Selected` enum('true','false') DEFAULT NULL; 
+-- Add ScannerCandID column to files, back-populate the newly added column from parameter_file table, then add foreign key constraints 
+ALTER TABLE files ADD `ScannerID` int(10) unsigned default NULL;
+CREATE TEMPORARY TABLE ScannerIDs AS SELECT pf.FileID, pf.Value AS ScannerID FROM parameter_file AS pf LEFT JOIN parameter_type AS pt ON pf.ParameterTypeID=pt.ParameterTypeID WHERE pt.Name='ScannerID';
```

Comment:
Existing study will need to check if they have the permission to create temporary tables for the user id they will import this patch with.